### ISSUE TITLE
Angular DI should allow NON-DI constructor arguments that can have a default value.

### ISSUE THREAD

**Issue by andreElrico** [angular/angular#37306]
# 🚀 feature request

### Relevant Package

Angular DI

### Description

related: #25233 , #25395   also [stackoverflow](https://stackoverflow.com/questions/48571489/setting-a-default-value-for-an-injectable-class)

In TS we can have constructor args with default value:

from [STACKBLITZ](https://stackblitz.com/edit/typescript-yjxrbn)

```
class HEHE {
  constructor(private val = 'lol-default') {
    console.log(this.val)
  }
}

new HEHE()

new HEHE('lol-custom')
```

However this behavior is DESTOYED by adding an angular decorator and therefore working with DI.

from [STACKBLITZ](https://stackblitz.com/edit/angular-ivy-gj8osq?file=src%2Fapp%2Fhehe.service.ts)

```
const MY_DEFAULT_VAL = "lol-default";

@Injectable()
export class HEHEService {

  constructor(
    @Inject(NgZone) private zone ,
    private val = MY_DEFAULT_VAL,
  ) {
    console.log(val)
  }

}

// Error: This constructor was not compatible with Dependency Injection.
```

In order to make it work again we have to do something stupid like:

```
@Injectable()
export class HEHEService {

  constructor(
    @Inject(NgZone) private zone ,
    @Optional() @Inject({howStupid: 'is that'}) val,
  ) {

    if (!val) val = MY_DEFAULT_VAL;
    console.log(val)
  }

}

@Injectable()
export class HEHEServiceEXTENDED extends HEHEService {

  constructor(
    @Inject(NgZone) zone: NgZone,
  ) {

    super(zone, 'why is angular DI destoying Typescript behavior ?')
    
  }

}
```

### Describe the solution you'd like

If DI is not "smart" enough to know that there is NOTHING to inject here:

`private val = MY_DEFAULT_VAL,`

there should at least be a decorator flag to opt out of DI:

`@Noop() private val = MY_DEFAULT_VAL,`

### Describe alternatives you've considered

A workaround is prominent in the second STACKBLITZ.

**Comment by pkozlowski-opensource** [comment]
@andreElrico I'm not sure how this one is different from the existing issue #25395 ? Isn't it a duplicate?

**Comment by andreElrico** [comment]
@pkozlowski-opensource the problem OP of  #25395 want so solve looks identical. however using `@optional` would mean we're already comitted to using the DI framework for this argument. My issue is to "neglect" DI for specific cases.

+ @Optional() means: dont throw an error when DI does not resolve
+ @Noop() would mean: dont use DI here at all. Use the normal typescript behaviour.
+ or DI would be "smart" enough to know when a "Noop" case is prominent.

**TLDR;**
if **TS**: `private val = MY_DEFAULT_VAL,`

works the same as

as **angulars** #25395: @Optional(MY_DEFAULT_VAL) private val,

Im okay in closing this issue into #25395. 

@pkozlowski-opensource pls decide yourself whether this issue has something new to offer or if it should be closed into #25395.

**Comment by kara** [comment]
@andreElrico I think it's better to combine this issue with #25395 because the root cause is the same (even though the suggested solution is different), so we can ensure that we don't try to solve these 2 things independently. But thank you for the detailed issue report. I'll comment on that thread to make sure your comments aren't lost.
